src/test/test_addrs.cc

```diff
@@ -237,3 +238,30 @@ TEST(Msgr, TestAddrvecEncodeAddrDecode3)
   ASSERT_NE(addr, addrvec.v[0]); // it's not the first addr(which is non-legacy)
   ASSERT_NE(addr, entity_addr_t(1, 0)); // it's not a blank addr either
 }
+
+const char *addrvec_parse_checks[][3] = {
+  { "127.0.0.1", "127.0.0.1:0/0", "" },
+  { "127.0.0.1 foo", "127.0.0.1:0/0", " foo" },
+  { "127.0.0.1 1.2.3.4 foo", "[127.0.0.1:0/0,1.2.3.4:0/0]", " foo" },
+  { "127.0.0.1 :: - foo", "[127.0.0.1:0/0,[::]:0/0,-]", " foo" },
+  { NULL, NULL, NULL },
+};
+
+TEST(entity_addrvec_t, parse)
+{
+  entity_addrvec_t addrvec;
+  bufferlist bl;
+
+  for (auto v : { addr_checks, addr_checks2, addrvec_parse_checks }) {
+    for (unsigned i = 0; v[i][0]; ++i) {
+      const char *end = "";
+      addrvec.parse(v[i][0], &end);
```

Comment:
could 
```c
ASSERT_TRUE(addrvec.parse(v[i][0], &end));
```
here

Reply:
but all of them contain one or more addresses in it. in other words, none of the test cases just contains invalid address.

Reply:
the 1.2.3 case didn't.  i added a return value check though that was also conditional on out.empty()